slack-api-client/src/test/java/test_with_remote_apis/scim/ApiTest.java

```diff
@@ -165,6 +186,78 @@ public void createAndDeleteUser() throws Exception {
         }
     }
 
+    @Test
+    public void createAndDeleteGuestUser() throws Exception {
+        if (orgAdminToken != null) {
+            UsersCreateResponse creation = createNewGuestUser();
+            assertThat(creation.getId(), is(notNullValue()));
+
+            UsersReadResponse readResp = slack.scim(orgAdminToken).readUser(req -> req.id(creation.getId()));
+            assertThat(readResp.getId(), is(creation.getId()));
+            assertThat(readResp.getUserName(), is(creation.getUserName()));
+            assertThat(readResp.getSlackGuest().getType(), is(User.SlackGuest.Types.MULTI));
+            assertThat(readResp.getSlackGuest().getExpiration(), is(notNullValue()));
+
+            try {
```

Comment:
What is the purpose of wrapping this test with try/finally, but with no catch? Would this lead to possibly some uncaught exceptions / silent test failures happening here? Looking through the CI logs, it seems difficult to determine if this test ran successfully or not.

Reply:
Just to clean up the data even when an exception is thrown from the search / update / patch operations.

> Looking through the CI logs, it seems difficult to determine if this test ran successfully or not.

As you can see, there are two top-level package names -- `test_locally` and `test_wth_remote_apis`. The CI builds do not run the ones under `test_wth_remote_apis`, that require real orgs/workspaces and tokens as of today. We run those tests every week day (by me) and I verify if they work when making PRs and merging PRs from external contributors.